### PR TITLE
Fix crash when a type-annotated `__slots__` with no value is declared.

### LINKED ISSUE #7280: Crash when providing type annotation for class __slots__ without assigning a value
### Bug description

Pylint crashes fatally when linting a file with the following example code:

example.py

```python
# pylint: disable=missing-docstring

class Foo:
    __slots__: str
```

pylint crashed with a ``AstroidError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "c:\Users\User\example\venv\lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 2663, in igetattr
    attributes = self.getattr(name, context, class_context=class_context)
  File "c:\Users\User\example\venv\lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 2603, in getattr
    raise AttributeInferenceError(
astroid.exceptions.AttributeInferenceError: '__slots__' not found on <ClassDef.Foo l.2 at 0x13232ba3220>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 727, in _check_file
    check_astroid_module(ast_node)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 926, in check_astroid_module
    retval = self._check_astroid_module(
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 976, in _check_astroid_module
    walker.walk(node)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\utils\ast_walker.py", line 93, in walk
    self.walk(child)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\utils\ast_walker.py", line 90, in walk
    callback(astroid)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\checkers\classes\class_checker.py", line 838, in visit_classdef
    self._check_slots(node)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\checkers\classes\class_checker.py", line 1369, in _check_slots
    for slots in node.igetattr("__slots__"):
  File "c:\Users\User\example\venv\lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 2714, in igetattr
    raise InferenceError(
astroid.exceptions.InferenceError: '__slots__' not found on <ClassDef.Foo l.2 at 0x13232ba3220>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 684, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 729, in _check_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```

A hotfix to my code is simply to assign a value to `__slots__` in the class definition.

Expecting pylint not to crash and an empty output or warning/error telling that I am using a defined but unassigned `__slots__`.

### Configuration

_No response_

### Command used

```shell
pylint example.py
```

### Pylint output

In the editor:

```shell
c:\Users\User\example\example.py: Fatal error while checking 'c:\Users\User\example\example.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in 'C:\Users\User\AppData\Local\pylint\pylint\Cache\pylint-crash-2022-08-09-13.txt'.
```

pylint-crash-2022-08-09-13.txt (and when running `pylint example.py`):

```
Traceback (most recent call last):
  File "c:\Users\User\example\venv\lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 2663, in igetattr
    attributes = self.getattr(name, context, class_context=class_context)
  File "c:\Users\User\example\venv\lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 2603, in getattr
    raise AttributeInferenceError(
astroid.exceptions.AttributeInferenceError: '__slots__' not found on <ClassDef.Foo l.2 at 0x13232ba3220>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 727, in _check_file
    check_astroid_module(ast_node)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 926, in check_astroid_module
    retval = self._check_astroid_module(
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 976, in _check_astroid_module
    walker.walk(node)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\utils\ast_walker.py", line 93, in walk
    self.walk(child)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\utils\ast_walker.py", line 90, in walk
    callback(astroid)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\checkers\classes\class_checker.py", line 838, in visit_classdef
    self._check_slots(node)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\checkers\classes\class_checker.py", line 1369, in _check_slots
    for slots in node.igetattr("__slots__"):
  File "c:\Users\User\example\venv\lib\site-packages\astroid\nodes\scoped_nodes\scoped_nodes.py", line 2714, in igetattr
    raise InferenceError(
astroid.exceptions.InferenceError: '__slots__' not found on <ClassDef.Foo l.2 at 0x13232ba3220>.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 684, in _check_files
    self._check_file(get_ast, check_astroid_module, file)
  File "c:\Users\User\example\venv\lib\site-packages\pylint\lint\pylinter.py", line 729, in _check_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```

### Expected behavior

Expected: pylint does not crash and either empty output, or some kind of warning or error that I'm defining but not assigning to a `__slots__` attribute.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.10.4 (tags/v3.10.4:9d38120, Mar 23 2022, 23:13:41) [MSC v.1929 64 bit (AMD64)]
```

### OS / Environment

_No response_

### Additional dependencies

astroid==2.11.7
colorama==0.4.5
dill==0.3.5.1
isort==5.10.1
lazy-object-proxy==1.7.1
mccabe==0.7.0
platformdirs==2.5.2
pylint==2.14.5
tomli==2.0.1
tomlkit==0.11.2
wrapt==1.14.1

### PR DESCRIPTION
**Pull request by mbyrnepr2**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

## Description

Fix crash when a type-annotated `__slots__` with no value is declared.

Closes #7280

### CODE REVIEW

File: tests/functional/s/slots_checks.py [hunk 1]
```diff
@@ -63,6 +63,13 @@ class NinthBad(object):
 class TenthBad(object):
     __slots__ = [1 + 2 + 3]  # [invalid-slots-object]
 
+class EleventhBad:  # [invalid-slots]
+    """type-annotated __slots__ with no value"""
+    __slots__: str
```
Comment (jacobtylerwalls) [inline tests/functional/s/slots_checks.py:68]: This is superfluous, but not invalid, right? Is there a good reason to emit `invalid-slots`?

**Review by jacobtylerwalls — CHANGES_REQUESTED** [review]
Thanks for jumping on a fix. I like the approach; it reduces complexity to iterate less, and I don't see a reason why we need to support redefined `__slots__` inside a class (famous last words...)

One question about whether to emit a message for the original example.

File: tests/functional/s/slots_checks.py (hunk 1 above)
Comment (mbyrnepr2) [reply tests/functional/s/slots_checks.py]: Good point @jacobtylerwalls!
Perhaps separately we can deal with this if we want to. In the meantime we can ignore this & focus on the crash aspect itself?

**Comment by jacobtylerwalls** [comment]
Update looks good. I realized why the iteration might be helpful, though.

Without iterating, this PR no longer emits `invalid-slots`:
```
class A:
    x = 1

    if x:
      __slots__ = ('1',)
    else:
      __slots__ = None
```

**Comment by mbyrnepr2** [comment]
> Update looks good. I realized why the iteration might be helpful, though.
> 
> Without iterating, this PR no longer emits `invalid-slots` for this example:
> 
> ```python
> class A:
>     x = 1
> 
>     if x:
>         __slots__ = ('x',)
>     else:
>         __slots__ = None
> ```

I see your point. I could be missing something but in that case I don't think we can use `igetattr` because if we have a class with multiple `__slots__` declared & one of them has an annotation with no value, we get the AttributeError/InferenceError - So we can't catch that one problematic case and then check the next items; see:
https://github.com/PyCQA/astroid/blob/main/astroid/nodes/scoped_nodes/scoped_nodes.py#L2635 which calls:
https://github.com/PyCQA/astroid/blob/main/astroid/nodes/scoped_nodes/scoped_nodes.py#L2571.

I don't see it used much in the codebase, but `ilookup` seems to do the trick; is that an ok approach?
```python
-> for slots in node.ilookup("__slots__"):
(Pdb) node.ilookup("__slots__")
<generator object _infer_stmts at 0x10a6987b0>
(Pdb) [s for s in node.ilookup("__slots__")]
[Uninferable, <Const.str l.8 at 0x10a3b66e0>]
(Pdb) 
```

**Review by jacobtylerwalls — APPROVED** [review]
Thank you! The smaller diff is terrific. I shudder to call some of these examples "Good" (lol) but I get why you needed to name them like that :-)

**Pull request merged**